src/Expression/Evaluator.purs

```diff
@@ -9,27 +9,27 @@ import Expression.Error (Expect, evaluationError, multipleErrors, throw, unknown
 import Expression.Syntax (BinaryOperation(..), Expression(..), UnaryOperation(..))
 import Expression.VariableMap (VariableMap, lookup)
 import IntervalArith.Approx (Approx, fromRationalPrec)
+import IntervalArith.Approx.Pi (piA)
 import IntervalArith.Approx.Sqrt (sqrtA)
 import IntervalArith.Misc (Rational, multiplicativePowerRecip, rationalToNumber)
-import Math (cos, exp, log, pow, sin, sqrt, tan, e, pi)
-
-presetConstants :: Array (Tuple String Number)
-presetConstants = [ (Tuple "pi" pi), (Tuple "e" e) ]
+import Math (cos, exp, log, pow, sin, sqrt, tan, pi)
 
 ----------------------------------------------------
 -- Evaluate Number
 ----------------------------------------------------
 roughEvaluate :: VariableMap Number -> Expression -> Expect Number
 roughEvaluate variableMap = case _ of
   ExpressionLiteral value -> pure $ rationalToNumber value
-  ExpressionVariable name -> case lookup variableMap name of
+  ExpressionVariable name -> case lookup (presetConstants <> variableMap) name of
     Just value -> pure value
     _ -> unknownValue name
   ExpressionBinary operation leftExpression rightExpression -> roughEvaluateBinaryOperation operation variableMap leftExpression rightExpression
   ExpressionUnary operation expression -> roughEvaluateUnaryOperation operation variableMap expression
   ExpressionLet name expression parentExpression -> case roughEvaluate variableMap expression of
     Right value -> roughEvaluate (variableMap <> [ (Tuple name value) ]) parentExpression
     Left error -> Left error
+  where
+  presetConstants = [ (Tuple "pi" pi) ]
```

Comment:
How come you removed `e` from the rough plot?

Reply:
I plan to put it back when we have it also in the robust evaluate.